core/src/services/zookeeper/backend.rs

```diff
@@ -0,0 +1,214 @@
+        }
+        self
+    }
+
+    pub fn user(&mut self, user: &str) -> &mut Self {
+        if !user.is_empty() {
+            self.user = Some(user.to_string());
+        }
+        self
+    }
+
+    pub fn password(&mut self, password: &str) -> &mut Self {
+        if !password.is_empty() {
+            self.password = Some(password.to_string());
+        }
+        self
+    }
+}
+
+impl Debug for ZookeeperBuilder {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("Builder");
+        if let Some(endpoint) = self.endpoint.clone() {
+            ds.field("endpoint", &endpoint)
+        }
+        if let Some(user) = self.user.clone() {
+            ds.field("user", &user)
+        }
+        if let Some(password) = self.password.clone() {
+            ds.field("password", &password)
+        }
+        ds.finish()
+    }
+}
+
+impl Builder for ZookeeperBuilder {
+    const SCHEME: Scheme = Scheme::Zookeeper;
+    type Accessor = ZookeeperBackend;
+
+    fn from_map(map: HashMap<String, String>) -> Self {
+        let mut builder = ZookeeperBuilder::default();
+
+        map.get("endpoint").map(|v| builder.endpoint(v));
+        map.get("user").map(|v| builder.user(v));
+        map.get("password").map(|v| builder.password(v));
+
+        builder
+    }
+
+    fn build(&mut self) -> Result<Self::Accessor> {
+        let endpoint = match self.endpoint.clone() {
+            None => DEFAULT_ZOOKEEPER_ENDPOINT,
+            Some(endpoint) => endpoint,
+        };
+
+        Ok(Backend::new(ZkAdapter {
+            endpoint: endpoint.to_string(),
+            user: self.user.clone(),
+            password: self.password.clone(),
+            client: OnceCell::new(),
+        }))
+    }
+}
+
+pub type Backend = kv::Backend<ZkAdapter>;
+
+#[derive(Clone)]
+pub struct ZkAdapter {
+    endpoint: String,
+    user: Option<String>,
+    password: Option<String>,
+    client: OnceCell<zk::Client>,
+}
+
+impl Debug for ZkAdapter {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("Adapter");
+        ds.field("endpoint", &self.endpoint);
+        ds.field("user", &self.user);
+        ds.field("password", &self.password);
+        ds.finish()
+    }
+}
+
+impl ZkAdapter {
+    async fn get_connection(&self) -> Result<zk::Client> {
+        if let Some(client) = self.client.get() {
+            return Ok(client.clone());
+        }
+        match zk::Client::connect(endpoint).await {
+            Ok(client) => {
+                match (self.user.clone(), self.password.clone()) {
+                    (Some(user), Some(password)) => {
+                        let auth = format!("{user}:{password}").as_bytes();
+                        client.auth(ZOOKEEPER_AUTH_SCHEME.to_string(), auth.to_vec());
+                    }
+                    _ => log::warn!("username and password isn't set, default use `anyone` acl"),
```

Comment:
OpenDAL prefer to use `warn!()` instead of `log::warn!()`. And using `debug` level here is enough, it's not a warning that users need to take action.